File: BE/src/main/java/com/ssafy/e106/global/common/ApiResponse.java
package com.ssafy.e106.global.common;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ApiResponse<T> {

  private final boolean success;
  private final T data;
  private final String message;

  public static <T> ApiResponse<T> ok(T data) {
    return new ApiResponse<>(true, data, null);
  }

  public static <T> ApiResponse<T> ok(T data, String message) {
    return new ApiResponse<>(true, data, message);
  }

  public static ApiResponse<Void> ok() {
    return new ApiResponse<>(true, null, null);
  }
}

File: FE/app/src/main/java/com/ssafy/e106/core/model/ApiResponse.kt
package com.ssafy.e106.core.model

import kotlinx.serialization.Serializable

@Serializable
data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val message: String? = null
)

@Serializable
data class ErrorResponse(
    val success: Boolean,
    val code: String,
    val message: String
)
